pkg/controller/status/status.go

```diff
@@ -359,19 +340,70 @@ func (c *Controller) merge(existing *configv1.ClusterOperator) *configv1.Cluster
 			Message: "Monitoring the cluster",
 		})
 	}
+}
 
-	if release := os.Getenv("RELEASE_VERSION"); len(release) > 0 {
-		existing.Status.Versions = []configv1.OperandVersion{
-			{Name: "operator", Version: release},
-		}
+func setInsightsDownloadDegradedOperatorStatusCondition(existing *configv1.ClusterOperator, last time.Time, downloadReason string, downloadMessage string) {
+	if len(downloadReason) > 0 {
+		setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+			Type:               InsightsDownloadDegraded,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: last},
+			Reason:             downloadReason,
+			Message:            downloadMessage,
+		})
+	} else {
+		removeOperatorStatusCondition(&existing.Status.Conditions, InsightsDownloadDegraded)
 	}
+}
 
-	if data, err := json.Marshal(reported); err != nil {
-		klog.Errorf("Unable to marshal status extension: %v", err)
+func setUploadDegradedOperatorStatusCondition(existing *configv1.ClusterOperator, last time.Time, uploadErrorReason string, uploadErrorMessage string) {
+	if len(uploadErrorReason) > 0 {
+		setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+			Type:               UploadDegraded,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: last},
+			Reason:             uploadErrorReason,
+			Message:            uploadErrorMessage,
+		})
 	} else {
-		existing.Status.Extension.Raw = data
+		removeOperatorStatusCondition(&existing.Status.Conditions, UploadDegraded)
+	}
+}
+
+func setDegradedOperatorStatusCondition(existing *configv1.ClusterOperator, last time.Time, errorReason string, errorMessage string,) {
+	if len(errorMessage) > 0 {
+		klog.V(4).Infof("The operator has some internal errors: %s", errorMessage)
+		setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+			Type:               configv1.OperatorDegraded,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: last},
+			Reason:             errorReason,
+			Message:            errorMessage,
+		})
+	} else {
+		setOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+			Type:   configv1.OperatorDegraded,
+			Status: configv1.ConditionFalse,
+			Reason: "AsExpected",
+		})
+	}
+}
+
+func setDisabledOperatorStatusCondition(existing *configv1.ClusterOperator, disabledMessage string, disabledReason string) {
```

Comment:
These new functions (with `setUploadDegradedOperatorStatusCondition`,`setDegradedOperatorStatusCondition`,`setInsightsDownloadDegradedOperatorStatusCondition`) are basically the same. What about to use only one or just create the corresponding `ClusterOperatorStatusCondition` first and then only pass it to `setOperatorStatusCondition` function.

Reply:
Yep! Thanks. This is probably the next step. I'm trying to do something about the `*reasons` and `*messages` to make it a little bit clean before cleaning up those methods.